qa/workunits/fs/snaps/snaptest-name-limits.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+#
+# This tests snapshot names limits: names have to be < 240 chars
+#
+
+function cleanup ()
+{
+	rmdir d1/.snap/*
+	rm -rf d1
+}
+
+function fail ()
+{
+	echo $@
+	cleanup
+	exit 1
+}
+
+mkdir -p d1/d2/d3/d4
+mkdir d1/.snap/snap1
+[ -d d1/.snap/snap1 ] || fail "Can't create snapshots"
+
+longname=$(printf "%.239d" 0)
+mkdir d1/.snap/$longname
+longsnap=$(ls d1/d2/d3/d4/.snap/ | grep $longname)
+echo "Snapshot name length: ${#longname} => Long name: ${#longsnap}"
+[ ${#longsnap} -eq 254 ] || fail "Failed to create snapshot with 239 chars"
+
+longname=$(printf "%.240d" 1)
+mkdir d1/.snap/$longname
+longsnap=$(ls d1/d2/d3/d4/.snap/ | grep $longname)
+echo "Snapshot name length: ${#longname} => Long name: ${#longsnap}"
+[ ${#longsnap} -eq 255 ] || fail "Failed to create snapshot with 240 chars"
+
+longname=$(printf "%.241d" 2)
+mkdir d1/.snap/$longname 2> /dev/null
+exists=$(ls d1/.snap|grep $longname)
+[ -z $exists ] || fail "Invalid snapshot exists: $exists"
```

Comment:
@luis-henrix I think we need only the last test (with 241), as the shorter name length test already covered somewhere else? If it errors out, the test is success.

Reply:
Ok, thanks.  I'll remove everything else from the script and push it the new version in a bit.